### DISCUSSION TITLE
Collection Quota

### DISCUSSION THREAD

**Discussion by mykgspace** [seaweedfs/seaweedfs#4359] [General]
Hello! I'm trying to give the client a certain number of volumes inside the collection, for this I enter the command with the -collectionQuotaMB flag:
 `weed mount -filer=localhost:8888 -filer.path=/test/ -dir=/mnt -collection=test -collectionQuotaMB=9216`

However, despite the quota, the client can write more data into the space allocated to him than the quota suggests.

Example. The client writes data, according to the quota, he still has 100 MB, he tries to write another 500 MB of data and the system allows him to do this, increasing the quota by 400 MB, and only after that says that there is no more space.

Tell me, please, is this a bug in the system or is there some other setting that makes quota control for the client more stringent?

Thank you in advance!

**Comment by chrislusf** [comment]
If the 500MB data comes in one file, or comes quickly within a minute, the quota could be exceeded as you described.

The quota is asynchronously enforced, for better performance purpose.

Even if enforced during write time, it is still possible that 2 or more concurrent write requests are trying to write at the same time, where the sum of all the requests may exceed the quota. So this would need a global lock on each write request and would slow down all write requests.

**Comment by mykgspace** [comment]
Thank you very much for the answer!)

I give the client a specific section (collection) and only he works with it. I need to make sure that the data allocated to it does not exceed a certain quota.
This mechanism, as far as I've seen, is provided at the level of the total disk space (-minFreeSpace).
Therefore, it is interesting to have a similar mechanism for collections, which would severely limit data writing when the limit is reached and transfer all data on disk to read-only.
